// react-search-autocomplete/dist/Results/Results.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Results;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _SearchIcon = require("../icons/SearchIcon");

var _StyledResults = require("./StyledResults");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Results(props) {
  var results = props.results,
      onClick = props.onClick,
      setSearchString = props.setSearchString,
      showIcon = props.showIcon,
      maxResults = props.maxResults;

  var handleClick = function handleClick(result) {
    onClick(result);
    setSearchString(result.name);
  };

  if (results.length <= 0) {
    return null;
  }

  return /*#__PURE__*/_react["default"].createElement(_StyledResults.StyledResults, null, /*#__PURE__*/_react["default"].createElement("div", {
    className: "line"
  }), /*#__PURE__*/_react["default"].createElement("ul", null, results.slice(0, maxResults).map(function (result) {
    return /*#__PURE__*/_react["default"].createElement("li", {
      key: "rsa-result-".concat(result.id),
      onMouseDown: function onMouseDown() {
        return handleClick(result);
      },
      onClick: function onClick() {
        return handleClick(result);
      }
    }, showIcon && /*#__PURE__*/_react["default"].createElement("div", {
      className: "icon"
    }, /*#__PURE__*/_react["default"].createElement(_SearchIcon.SearchIcon, null)), /*#__PURE__*/_react["default"].createElement("div", {
      className: "ellipsis",
      title: result.name
    }, result.name));
  })));
}

Results.defaultProps = {
  results: [],
  setSearchString: function setSearchString() {}
};
Results.propTypes = {
  results: _propTypes["default"].array,
  onClick: _propTypes["default"].func,
  setSearchString: _propTypes["default"].func,
  showIcon: _propTypes["default"].bool,
  maxResults: _propTypes["default"].number
};
